### DISCUSSION TITLE
How to set different colors for dark and light mode?

### DISCUSSION THREAD

**Discussion by MGeurts** [tasteui/tasteui#376] [Q&A] [answered]
How can we define a seperate color for **dark mode** and **light mode** without writing if - then - else statements for every component?

The developers dicided to take the color option out of the component class and into its own attribute **color**.

The following does not work:

`<x-ts-button color="secondary" dark:color="red">Button 1</x-ts-button>`

`<x-ts-button color="secondary dark:red">Button 2</x-ts-button>`

Please help. Thanks.

**Comment by devajmeireles** [comment, marked as answer]
Hey, @MGeurts !

This is something I'm thinking about. We currently don't have a way to set the dark color based on your example, but I'm thinking about what the best approach for this will be, and then I'll build the feature.

**Reply by MGeurts** [reply]
OK. Thanks.

**Comment by MGeurts** [comment]


**Reply by devajmeireles** [reply]
Can you say what is difficult for you to understand? I've invested a lot of time into completing the documentation, and I'm still working on it, so I welcome your feedback to correct anything wrong or difficult to understand.

### Soft personalization:
- Way to touch the components' classes (not colors) without needing a lot of technical knowledge. Every component has its CSS classes exported from PHP through named blocks. In the documentation for each component, there will be a button that, when clicked, will display a modal with the blocks and original contents. Logically you need to pay attention to customize things, but it's not difficult. In this code below I am making all my inputs fully rounded:

Before:
![CleanShot 2024-03-29 at 17 49 59@2x](https://github.com/tallstackui/tallstackui/assets/60591772/b2d0d299-08a3-47a2-801b-016fe2b958f7)

After:
![CleanShot 2024-03-29 at 17 51 33@2x](https://github.com/tallstackui/tallstackui/assets/60591772/8bf51afe-0032-43c9-b8c3-22347c027837)

Code:
```php
namespace App\Providers;

use Illuminate\Support\ServiceProvider;
use TallStackUi\Facades\TallStackUi;

class AppServiceProvider extends ServiceProvider
{
    public function boot(): void
    {
        TallStackUi::personalize()
            ->form('input')
            ->block('input.wrapper')
            ->replace('rounded-md', 'rounded-full');
    }

    public function register(): void
    {
        //
    }
}
```

---

### Deep personalization:
- It requires more technical knowledge because it requires overwriting the original class of each component, and that's where you have to work to create custom colors. In this example, I'm creating a color called "ocean" (it doesn't exist in TailwindCSS):

Codes:

1. In the config file:

```php
// ...

    'components' => [
        'alert' => Components\Alert::class,
        'avatar' => Components\Avatar::class,
        'badge' => Components\Badge::class,
        'banner' => Components\Banner::class,
        //'button' => Components\Button\Button::class,
        'button' => App\View\Components\Button::class,
// ...
```

2. The component PHP class:

```php
namespace App\View\Components;

class Button extends \TallStackUi\View\Components\Button\Button
{
    public function backgroundColor(): array
    {
        return [
            'solid' => [
                // other indexes goes here ...
 
                'ocean' => 'bg-ocean-500 border-ocean-500', 
            ],
            'outline' => [
                // other indexes goes here
 
                'ocean' => 'bg-transparent border-ocean-500', 
            ],
            'light' => [
                // other indexes goes here
 
                'ocean' => 'bg-ocean-300 border-ocean-300', 
            ],
        ];
    }
}
```

3. The `tailwind.config.js` file:

```js
import defaultTheme from 'tailwindcss/defaultTheme';
import forms from '@tailwindcss/forms';

/** @type {import('tailwindcss').Config} */
export default {
    darkMode: 'class',

    presets: [
        require('./vendor/tallstackui/tallstackui/tailwind.config.js')
    ],

    content: [
        './vendor/laravel/framework/src/Illuminate/Pagination/resources/views/*.blade.php',
        './storage/framework/views/*.php',
        './resources/views/**/*.blade.php',
        //
        './vendor/tallstackui/tallstackui/src/**/*.php',
        './app/View/**/*.php',
        './app/Providers/AppServiceProvider.php'
    ],

    theme: {
        extend: {
            fontFamily: {
                sans: ['Figtree', ...defaultTheme.fontFamily.sans],
            },
            colors: {
                'ocean': {
                    '50': '#f1f8fe',
                    '100': '#e3effb',
                    '200': '#c0e0f7',
                    '300': '#88c7f1',
                    '400': '#49aae7',
                    '500': '#2292d8',
                    '600': '#1472b5',
                    '700': '#115b93',
                    '800': '#124e7a',
                    '900': '#154265',
                    '950': '#0e2a43',
                },
            }
        },
    },

    plugins: [forms],
};
```

Result:
![CleanShot 2024-03-29 at 17 57 58@2x](https://github.com/tallstackui/tallstackui/assets/60591772/293ecf74-835f-4fe9-8647-1f8150303024)

---

All of these examples were based on the documents. Could you tell me what is difficult to understand in the docs?

**Comment by MGeurts** [comment]
OK. Slowly it starts to get clearer.

My first mistake was that I did not recognize the button **Personalize: Component** as a button in the **Soft Personalization** section. I thought it was a label so I never saw the documentation behind it. Stupid me. I apologize…

Second the **Deep Personalization** example above is just a little bit more specific to the problem I had, so this time I understood the workflow. 

I can clearly see you put a lot of effort into this library. I really appreciate it. Thank you very much.

**Reply by devajmeireles** [reply]
I'm writing improved documentation and making some topics even easier to understand. I'd love to hear your feedback on what should improve. Although the current documentation is effective, it could improve in some aspects. Overall, I suggest you read carefully.
